extensions/project-descriptor.md

```diff
@@ -0,0 +1,137 @@
+# Project Descriptor
+
+A project descriptor is a file that contains configuration for apps, services, functions, and buildpacks. By default, the file is named `project.toml` and located in the root directory of a project's repository.
+
+## Schema
+
+The TOML schema of the project descriptor is the following:
+
+```toml
+[project]
+id = "<string>" # machine readble
+name = "<string>" # human readable
+version = "<string>"
+authors = ["<string>"]
+documentation-url = "<url>"
+source-url = "<url>"
+
+[[project.licenses]]
+type = "<string>"
+uri = "<uri>"
+
+[build]
+include = ["<string>"]
+exclude = ["<string>"]
+[[build.buildpacks]]
+id = "<string>"
+version = "<string>"
+uri = "<string>"
+[[build.env]]
+name = "<string>"
+value = "<string>"
+[metadata]
+# additional arbitrary keys allowed
+```
+
+The following sections describe each part of the schema in detail.
+
+## `[project]`
+
+The top-level `[project]` table may contain configuration about the repository, including `id` and `version`, but also metadata about how it is authored, documented, and version controlled.
+
+The `project.id`
+
+```toml
+[project]
+id = "<string>"
+name = "<string>"
+version = "<string>"
+```
+
+* `id` - the machine readable identifier of the project (ex. "com.example.myservice")
+* `name` - the human readable name of the project (ex. "My Example Service")
```

Comment:
This appears to be either missing some definition of optionality or the example at the bottom is incomplete.